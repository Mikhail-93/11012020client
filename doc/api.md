# Аутентификация
в каждом гет-запросе присутствуют ключи ?uid=xxx&hash=yyy. В каждом пост-запросе эти же ключи присутствуют в теле (перенести в HTTP-заголовок?). 
Правила поверки подленности описаны здесь https://vk.com/dev/Login
```
app_id+user_id+secret_key, например md5(194253766748fTanppCrNSeuYPbA4ENCo)
```

# Правила именования свойств
camelCase

# Свойства - объекты
Свойства, которые являются объектами, передаются иднтификатором объекта. Если поле null, то оно передается как null
```json
"witness_id":"8274",
"foreman_id":null
```
  
# Свойства - коллекции
Свойства, которые являются коллекциями, передаются null
```json
"ten": null
```  

# Ошибки
HTTP status code: 200 (как у всех успешных запросов)
```json
{
    "error": {
        "error_code": 113,
        "error_msg": "Invalid user id",
        "request_params": [{
            "key": "oauth",
            "value": "1"
            }, {
            "key": "method",
            "value": getInstancenstance"
            }, {
            "key": "fields",
            "value": "photo_50,city,verified"
            }, {
            "key": "name_case",
            "value": "Nom"
            }, {
            "key": "user_ids",
            "value": "210700286777"
            }, {
            "key": "v",
            "value": "5.103"
        }]
    }
}
```

# Получить нескольких пользовалетей
GET users/get?ids=210700286,7777777
status:2 можно не отправлять поле status
```json
{
    "response": [{
        "id": 210700286,
        "uid": 12454352,
        "lastName": "Stirling",
        "firstName": "Lindsey",
        "patronymic": "asdf",
        "nickname": "Boroda",
        "birthyear": "1900",
        "location": ["1234.3125","54.3245"], // latitude, longitude
        "photo": "https://sun1-19.u...EGxg5NXns.jpg?ava=1",
        "smallPhoto": "https://sun1-19.u...EGxg5NXns.jpg?ava=1",
        "status": 1, // const STATUS_NEW= 0;  STATUS_PENDING= 1; STATUS_CONFIRMED= 2; STATUS_DECLINE= 3;
        "witnessChatInviteLink": "vk.com/join........",
,
        "witness_id": "1234",
        "foreman_id": null,

        "ten": "undefined",
    },
    {
        "id": "7777777",
      ...
    }]
 }
 ```

# Получить пользователя по ВК uid
возвращает response: null, если соответствующего пользователя в Копнике не заведено
GET users/getByUid?uid=210700286

```json
{
    "response": {
      "id": 1034853,
      "firstName": "fasdfasdf",
      ... и т.д. все содержимое одного элемента из ответа users/get(ids)
    }
 }
```

# Получить count самых старших пользователей в заданном квадрате в порядке возрастния ранга
GET users/get_top_inside_square?x1=123.123,y1=123.123,x2=1243.24,y2=53.23,count=20

Ответ аналогичный методу getInstance


# Оставить заявку на регистрацию
Если заявка от этого пользователя ВК уже есть, заменить ее этой
PUT users/putWitnessRequest
```json
body:{
  "firstName": "sadgfd",
  "lastName": "jklsdfg89rfs",
  "patronymic": "sdfafasd",
  "birthyear": "1900",
  "passport": "1984",
  "nickname": "sdakljh23",
  "location": [1238974.12, 9124.12390]

  uid: 1244534,                //стандартная прибавка ко всем ПОСТ-запросам
  hash: "fJKDGL98&*#%ife"
}
``` 
