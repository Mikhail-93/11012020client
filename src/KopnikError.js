class KopnikError extends Error {
    constructor(message, code) {
        super(message)
        this.code = code
    }
}
class KopnikApiError extends KopnikError {
    constructor(message, code, url) {
        super(message, code)
        this.url = url
    }
}



export {KopnikError, KopnikApiError}
