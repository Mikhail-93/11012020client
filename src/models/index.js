/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}