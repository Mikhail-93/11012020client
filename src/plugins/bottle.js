import Bottle from "bottlejs"
import _ from "lodash"
import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../Application"
import config from '../../config'
import fetchApi from "../bottle/fetchApi";
import fetchApiMock from "../bottle/fetchApi.mock";

Bottle.config.strict = true
const bottle = new Bottle()

/**
 * Настраивает контейнер
 * Для тестов необходим сброс и перенастройка чтобы соблюсти изолированность тестов
 *
 * @param {Object} options параметры по умолчанию соответствуют наиболее приемлемым для окружения значениям окружению
 * @param {Number=} options.cookie
 * @param {Boolean=} options.fetch
 */
Bottle.prototype.setup = function (options) {
    bottle.factory('defaultFetchApiOptions', function defaultFetchApiOptionsFactory(container) {
        const cookies = {
            1: 'PHPSESSID=user-1',
            2: 'PHPSESSID=eqlae6a1grpvcirpemfvebf806',
            3: 'PHPSESSID=user-3',
            4: 'PHPSESSID=user-4',
            5: 'PHPSESSID=user-5',
            6: 'PHPSESSID=user-6',
            7: 'PHPSESSID=user-7',
            8: 'PHPSESSID=user-8',
            9: 'PHPSESSID=user-9',
            10: 'PHPSESSID=user-10',
        }
        const result = {}
        if (container.config.di.cookie) {
            result.headers = {
                cookie: cookies[container.config.di.cookie]
            }
        }
        // console.log(result)
        return result
    })
    bottle.factory('fetchApi', function fetchApiFactory(container) {
        return container.config.di.fetch ? fetchApi : fetchApiMock
    })
    bottle.factory('config', function configFactory() {
        if (!process.env.NODE_ENV) {
            throw new Error("NODE_ENV is not defined");
        }

        let local = {}//require("./local.js")
        let result = _.merge({}, config, local)[process.env.NODE_ENV]

        return result
    })
    if (!Application){
        throw new Error("Application is undefined in bottle")
    }
    bottle.service('application', Application, 'logger')
    bottle.factory('logger', function loggerFactory() {
        //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
        // LoglevelPluginPrefix.reg(loglevel)
        // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
        loglevel.setLevel(loglevel.levels.DEBUG)
        if (process.env.NODE_ENV == "test") {
            loglevel.setLevel(loglevel.levels.WARN)
            // LoglevelPluginToString.apply(loglevel, {})
        }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
        return loglevel
    })
    Object.assign(this.container.config.di, options)
}

bottle.setup()

/**
 * @callback fetch
 * @param {string} url
 * @param {Object} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} fetchApi
 * @property {Object} defaultFetchApiOptions
 * @property {Object} defaultFetchApiOptions.headers
 * @property {Object} defaultFetchApiOptions.headers.cookie
 * @property {Object} config
 */
const container = bottle.container
export {bottle, container}
