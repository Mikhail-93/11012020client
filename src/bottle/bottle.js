import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
