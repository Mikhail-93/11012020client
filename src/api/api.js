import {container} from "../bottle/bottle";
import {KopnikApiError} from "../KopnikError";
import _ from "lodash";

export default async function api(url, options = {}) {
    const logger = container.logger.getLogger('api')
    let response

    // 1. setup options
    const fullOptions = _.merge({}, options, {
        method: (options.method || 'GET').toUpperCase(),
        credentials: 'include',
        headers: {
            Cookie: container.constants.di.cookie ? container.cookieService.cookie : undefined,
            Accept: 'application/json',
            'Content-Type': options.method === 'GET' ? 'text/plain' : 'application/x-www-form-urlencoded;charset=UTF-8',
        },
        body: options.body ? JSON.stringify(options.body) : undefined,
    })

    //2. set full path to api endpoint
    let fullUrl = container.constants.api.path + '/' + url

    //3. execute fetch
    try {
        logger.debug(...[fullOptions.method, fullUrl, options.body, fullOptions.headers.Cookie].filter(item => item))
        response = await fetch(fullUrl, Object.assign({api: true}, fullOptions))
    } catch (err) {
        // abort getTopInsideSquare for example
        if (err.name == 'AbortError') {
            err.url = fullUrl
            throw err
            // miss network
        } else {
            throw new KopnikApiError(err.message, err.code, fullUrl)
        }
    }
    // 4. set cookie
    if (container.constants.di.cookie) {
        let cookie = response.headers.get('set-cookie')
        if (!cookie) {
            // logger.warn('no cookie received from server')
        } else {
            cookie = cookie.match(/(\w+=(\w|\d)+)/)[0]
            container.cookieService.cookie = cookie
            // logger.debug('receive', cookie)
        }
    }
    // 5. parse response body
    const bodyText = await response.text()

    // 6. soft error
    if (!response.ok) {
        throw new KopnikApiError(`${response.statusText}. ${bodyText}`, response.status, fullUrl)
    }

    // 7. parse response body as JSON
    let body
    try {
        body = JSON.parse(bodyText)
    } catch (err) {
        throw new KopnikApiError(bodyText, 2000001, fullUrl)
    }

    // 8. check error from server
    if (body.error) {
        throw new KopnikApiError(body.error.error_msg, body.error.error_code, fullUrl)
    } else {
        logger.debug(...[body, response.headers.get('set-cookie')].filter(item=>item))
    }

    // return result
    return body.response
}
