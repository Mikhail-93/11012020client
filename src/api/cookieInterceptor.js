import {container} from "../plugins/bottle";
import _ from 'lodash'

export default {
    request: function (url, options) {
        if (!url.includes(container.config.api.path)) {
            return [url, options]
        }
        if (!container.config.di.cookie) {
            return [url, options]
        }
        _.merge(options, {
            headers: {
                Cookie: container.cookieService.cookie
            }
        })

        return [url, options]
    },
    requestError: function (error) {
        return Promise.reject(error);
    },
    response: async function (response) {
        if (!response.url.includes(container.config.api.path)) {
            return response
        }
        if (!container.config.di.cookie) {
            return response
        }
        let cookie = response.headers.get('set-cookie')
        if (!cookie){
            // container.logger.warn('no cookie received from server')
        }
        else {
            cookie = cookie.match(/(\w+=(\w|\d)+)/)[0]
            container.cookieService.cookie = cookie
        }
        return response
    },

    responseError: function (error) {
        return Promise.reject(error);
    }
}
