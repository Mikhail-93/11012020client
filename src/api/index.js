import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
