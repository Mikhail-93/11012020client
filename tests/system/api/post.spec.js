import api from "../../../src/api";
import {bottle, container} from "../../../src/bottle/bottle";
import {KopnikApiError} from "../../../src/KopnikError";
import {AbstractSync, Kopnik} from "../../../src/models";

/**
 * Реализация без участия контейнера
 * @param id
 * @returns {Promise<*|undefined>}
 */
function login(id) {
    return api('test/login/' + id)
}

container.config.di.fetch = true

describe('system api post', () => {
    beforeEach(async () => {
        AbstractSync.clearCache()
    })
    describe('anonymous', () => {
        beforeAll(() => {
            bottle.resetProviders(['cookieService'])
        })
        it('users/update', async () => {
            let kopnik = new Kopnik()
            try {
                await api('users/update', {
                    method: 'POST',
                    body: {
                        id: 2,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(401)
                expect(err).toMatchSnapshot()
            }
        })
        it('users/pending/update', async () => {
            let kopnik = new Kopnik()
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 1,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(401)
                expect(err).toMatchSnapshot()
            }
        })
    })
    /********************************************************
     * user_1_confirmed (witness)
     *******************************************************/
    describe('user1', () => {
        beforeAll(async () => {
            await login(1)
        })

        it('users/pending/update', async () => {
            let result = await api('users/pending/update', {
                method: 'POST',
                body: {
                    id: 3,
                    status: 2,
                }
            })
            let kopnik = await Kopnik.get(3)
            expect(kopnik.status).toBe(2)
            expect(result).toMatchSnapshot()
        })

        it('users/pending/update(new)', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 2,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(5)
            }
        })
        it('users/pending/update(declined)', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 4,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(5)
            }
        })
        it('users/pending/update(confirmed)', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 5,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(5)
            }
        })

    })
    /********************************************************
     * user_2_new
     *******************************************************/
    describe('user2', () => {
        beforeAll(async () => {
            await login(2)
        })
        it('users/update', async () => {
            const kopnik = await Kopnik.get(2)
            let result = await api('users/update', {
                method: 'POST',
                body: kopnik.plain
            })
            await kopnik.reload()
            expect(kopnik.status).toBe(1)
            expect(result).toMatchSnapshot()
        })
        it('users/pending/update', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 3,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(403)
                expect(err).toMatchSnapshot()
            }
        })
    })
    /********************************************************
     * user_3_pending
     *******************************************************/
    describe('user3', () => {
        beforeAll(async () => {
            await login(3)
        })
        it('users/update', async () => {
            const kopnik = await Kopnik.get(3)
            let result = await api('users/update', {
                method: 'POST',
                body: kopnik.plain,
            })
            await kopnik.reload()
            expect(kopnik.status).toBe(1)
            expect(result).toMatchSnapshot()
        })
        it('users/pending/update', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 3,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(403)
                expect(err).toMatchSnapshot()
            }
        })
    })
    /********************************************************
     * user_4_declined
     *******************************************************/
    describe('user4', () => {
        beforeAll(async () => {
            await login(4)
        })
        it('users/update', async () => {
            const kopnik = await Kopnik.get(4)
            let result = await api('users/update', {
                method: 'POST',
                body: kopnik.plain
            })
            await kopnik.reload()
            expect(kopnik.status).toBe(1)
            expect(result).toMatchSnapshot()
        })
        it('users/pending/update', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 3,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(403)
                expect(err).toMatchSnapshot()
            }
        })
    })
    /********************************************************
     * user_5_confirmed
     *******************************************************/
    describe('user5', () => {
        beforeAll(async () => {
            await login(5)
        })
        it('users/update', async () => {
            const kopnik = await Kopnik.get(5)
            let result = await api('users/update', {
                method: 'POST',
                body: kopnik.plain
            })
            await kopnik.reload()
            expect(kopnik.status).toBe(1)
            expect(result).toMatchSnapshot()
        })
        it('users/pending/update', async () => {
            try {
                await api('users/pending/update', {
                    method: 'POST',
                    body: {
                        id: 3,
                        status: 2,
                    }
                })
                throw new Error("should not be hire")
            } catch (err) {
                expect(err).toBeInstanceOf(KopnikApiError)
                expect(err.code).toBe(403)
                expect(err).toMatchSnapshot()
            }
        })
    })

    /********************************************************
     * user_6_foreman
     *******************************************************/
    describe('user6', () => {
        beforeAll(async () => {
            await login(6)
        })
    })
})

