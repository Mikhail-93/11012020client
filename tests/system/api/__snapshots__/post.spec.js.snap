// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`system api post anonymous users/pending/update 1`] = `{ error: {"message":"No authentication","code":401,"url":"http://localhost:8082/api/users/pending/update"} }`;

exports[`system api post anonymous users/update 1`] = `{ error: {"message":"No authentication","code":401,"url":"http://localhost:8082/api/users/update"} }`;

exports[`system api post user1 users/pending/update 1`] = `true`;

exports[`system api post user2 users/pending/update 1`] = `{ error: {"message":"ACCESS_DENIED: заявку могут обрабатывать только заверители","code":403,"url":"http://localhost:8082/api/users/pending/update"} }`;

exports[`system api post user2 users/update 1`] = `
Array [
  Object {
    "birthyear": 1961,
    "firstName": "Трофим",
    "foreman_id": null,
    "id": 1,
    "lastName": "Гурьев",
    "locale": "ru",
    "location": Object {
      "lat": 62.335961,
      "lng": 29.089971,
    },
    "passport": null,
    "patronymic": "Колесников",
    "photo": "https://lorempixel.com/200/200/?38191",
    "smallPhoto": "https://lorempixel.com/200/200/?38191",
    "status": 2,
    "witness_id": null,
  },
]
`;

exports[`system api post user3 users/pending/update 1`] = `{ error: {"message":"ACCESS_DENIED: заявку могут обрабатывать только заверители","code":403,"url":"http://localhost:8082/api/users/pending/update"} }`;

exports[`system api post user3 users/update 1`] = `
Array [
  Object {
    "birthyear": 1961,
    "firstName": "Трофим",
    "foreman_id": null,
    "id": 1,
    "lastName": "Гурьев",
    "locale": "ru",
    "location": Object {
      "lat": 62.335961,
      "lng": 29.089971,
    },
    "passport": null,
    "patronymic": "Колесников",
    "photo": "https://lorempixel.com/200/200/?38191",
    "smallPhoto": "https://lorempixel.com/200/200/?38191",
    "status": 2,
    "witness_id": null,
  },
]
`;

exports[`system api post user4 users/pending/update 1`] = `{ error: {"message":"ACCESS_DENIED: заявку могут обрабатывать только заверители","code":403,"url":"http://localhost:8082/api/users/pending/update"} }`;

exports[`system api post user4 users/update 1`] = `
Array [
  Object {
    "birthyear": 1961,
    "firstName": "Трофим",
    "foreman_id": null,
    "id": 1,
    "lastName": "Гурьев",
    "locale": "ru",
    "location": Object {
      "lat": 62.335961,
      "lng": 29.089971,
    },
    "passport": null,
    "patronymic": "Колесников",
    "photo": "https://lorempixel.com/200/200/?38191",
    "smallPhoto": "https://lorempixel.com/200/200/?38191",
    "status": 2,
    "witness_id": null,
  },
]
`;

exports[`system api post user5 users/pending/update 1`] = `{ error: {"message":"ACCESS_DENIED: заявку могут обрабатывать только заверители","code":403,"url":"http://localhost:8082/api/users/pending/update"} }`;

exports[`system api post user5 users/update 1`] = `
Array [
  Object {
    "birthyear": 1961,
    "firstName": "Трофим",
    "foreman_id": null,
    "id": 1,
    "lastName": "Гурьев",
    "locale": "ru",
    "location": Object {
      "lat": 62.335961,
      "lng": 29.089971,
    },
    "passport": null,
    "patronymic": "Колесников",
    "photo": "https://lorempixel.com/200/200/?38191",
    "smallPhoto": "https://lorempixel.com/200/200/?38191",
    "status": 2,
    "witness_id": null,
  },
]
`;
